nsdb-cluster/src/main/scala/io/radicalbit/nsdb/cluster/actor/ClusterListener.scala

```diff
@@ -87,28 +95,56 @@ class ClusterListener() extends Actor with ActorLogging {
 
   override def postStop(): Unit = cluster.unsubscribe(self)
 
+  protected def createNodeActorsGuardian(): ActorRef =
+    context.system.actorOf(
+      NodeActorsGuardian.props(self).withDeploy(Deploy(scope = RemoteScope(cluster.selfMember.address))),
```

Comment:
I was thinking, also given the comment left below, that perhaps it might be possible to pass the NodeActorsGuardian.props function reference as a constructor parameter of the cluster listener. In this way, we can avoid any related mocking. I'm not sure about serialization or other implications. You could give it a try anyway, just to prove (or not) my point

Reply:
we can make a try! For sure the testing part would be get easier!!